Google.PowerShell.Tests/Common/TestActiveUserCredential.cs

```diff
@@ -0,0 +1,138 @@
+        /// ActiveUserCredential is created.
+        /// </summary>
+        [Test]
+        public void TestActiveUserCredentialConstructor()
+        {
+            ActiveUserCredential activeUserCred = new ActiveUserCredential();
+            ActiveUserToken activeUserToken = TokenProperty.GetValue(null, null) as ActiveUserToken;
+            Assert.IsNotNull(activeUserToken, "ActiveUserToken should be created.");
+        }
+
+        /// <summary>
+        /// This test checks that a call to RefreshTokenAsync will give us a new token.
+        /// </summary>
+        [Test]
+        public void TestRefreshTokenAsync()
+        {
+            ActiveUserCredential activeUserCred = new ActiveUserCredential();
+            ActiveUserToken activeUserToken = TokenProperty.GetValue(null, null) as ActiveUserToken;
+
+            bool refreshed = activeUserCred.RefreshTokenAsync(CancelToken).Result;
+
+            Assert.IsTrue(refreshed, "RefreshTokenAsync should return true.");
+
+            ActiveUserToken refreshedActiveUserToken = TokenProperty.GetValue(null, null) as ActiveUserToken;
+
+            Assert.IsTrue(
+                !Equals(activeUserToken.AccessToken, refreshedActiveUserToken.AccessToken),
+                "Refreshed ActiveUserToken should have a different access token.");
+        }
+
+        /// <summary>
+        /// This test checks that GetAccessTokenForRequestAsync works.
+        /// </summary>
+        [Test]
+        public void TestGetAccessTokenForRequestAsync()
+        {
+            ActiveUserCredential activeUserCred = new ActiveUserCredential();
+            ActiveUserToken activeUserToken = TokenProperty.GetValue(null, null) as ActiveUserToken;
+
+            string accessToken = activeUserCred.GetAccessTokenForRequestAsync(null, CancelToken).Result;
+
+            // The access token returned by GetAccessTokenForRequestAsync should come from token.
+            Assert.IsTrue(
+                Equals(activeUserToken.AccessToken, accessToken),
+                "GetAccessTokenForRefreshAsync returns the wrong access token.");
+
+            // The next call to GetAccessTokenForRequestAsync should returns the same token.
+            accessToken = activeUserCred.GetAccessTokenForRequestAsync(null, CancelToken).Result;
+            Assert.IsTrue(
+                Equals(activeUserToken.AccessToken, accessToken),
+                "GetAccessTokenForRefreshAsync returns the wrong access token.");
+        }
+
+        /// <summary>
+        /// This test checks that GetAccessTokenForRequestAsync returns a new token
+        /// when the old one expired due to time.
+        /// </summary>
+        [Test]
+        public void TestGetAccessTokenExpiredByTime()
+        {
+            ActiveUserCredential activeUserCred = new ActiveUserCredential();
+            ActiveUserToken activeUserToken = TokenProperty.GetValue(null, null) as ActiveUserToken;
+
+            // Make the token expired by time.
```

Comment:
"expired by time" isn't especially cromulent. Perhaps "Force the access token to expire."